corehq/apps/data_analytics/gir_generator.py

```diff
@@ -0,0 +1,126 @@
+import datetime
+
+from collections import namedtuple
+
+from django_countries.data import COUNTRIES
+
+from dimagi.utils.dates import add_months
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.data_analytics.esaccessors import (
+    get_domain_device_breakdown_es, active_mobile_users, get_possibly_experienced
+)
+from corehq.apps.data_analytics.models import MALTRow, GIRRow
+from corehq.apps.data_analytics.const import (
+    TEST_COUCH_TO_SQL_MAP, AMPLIFY_COUCH_TO_SQL_MAP, NOT_SET, BU_MAPPING
+)
+
+
+UserCategories = namedtuple('UserCategories', 'active performing experienced total sms')
+
+
+class GIRTableGenerator(object):
+    """
+        Populates SQL table with data for given list of monthly-datespans
+        See .models.GIRRow
+    """
+
+    def __init__(self, datespan_object_list):
+        self.monthspan_list = datespan_object_list
+
+    def build_table(self):
+        rows = []
+        for domain in Domain.get_all():
+            for monthspan in self.monthspan_list:
```

Comment:
nit: this might be more readable if you pulled out the body of the loop into its own function, e.g. `get_gir_dict_for_domain_and_monthspan` or something